src/Tasks.UnitTests/ResourceHandling/GenerateResource_Tests.cs

```diff
@@ -3395,8 +3396,42 @@ public static void AssertLogContainsResource(GenerateResource t, string messageI
         {
             Assert.Contains(
                 String.Format(AssemblyResources.GetString(messageID), replacements),
-                ((MockEngine)t.BuildEngine).Log
-                );
+                ((MockEngine) t.BuildEngine).Log
+            );
+        }
+
+        /// <summary>
+        /// Looks for a formatted message in the output log for the task execution, with unknown formatted parameters.
+        /// If verifies that all constant segments of unformatted message are present.
+        /// </summary>
+        public static void AssertLogContainsResourceWithUnspecifiedReplacements(GenerateResource t, string messageID)
+        {
+            var unformattedMessage = AssemblyResources.GetString(messageID);
+            var matches = Regex.Matches(unformattedMessage, @"\{\d+.*?\}");
+            if (matches.Count > 0)
+            {
+                var sb = new StringBuilder();
+                int i = 0;
+
+                foreach (Match match in matches)
+                {
+                    string segment = unformattedMessage.Substring(i, match.Index - i);
+                    sb.Append(Regex.Escape(segment));
+                    sb.Append(".*");
+
+                    i = match.Index + match.Length;
+                }
+                if (i < unformattedMessage.Length)
+                {
+                    sb.Append(Regex.Escape(unformattedMessage.Substring(i)));
+                }
+
+                Assert.Matches(sb.ToString(), ((MockEngine)t.BuildEngine).Log);
```

Comment:
I'm not sure if there's a Shouldly equivalent of this, but it would be nice if there is.

Reply:
I wanted to keep it consistent with rest of similar methods. Having Utility method named Assert* which uses ShouldLy is confusing to me. If this is to be changed this Utility methods shall be reworked into ShouldLy naming standards in different PR.

Reply:
That's ok. I would argue that, unlike most code, the main way we would "look" at it is through the messages that come out, i.e., x should be y versus x and y should be equal. The former is clearer, but it isn't a huge deal.